company.el

```diff
@@ -2536,12 +2545,16 @@ character, stripping the modifiers.  That character must be a digit."
                         (car (last (string-to-list (symbol-name type))))))
                 (n (- char ?0)))
            (if (zerop n) 10 n))))
+  (company-complete-nth (cl-decf n)))
```

Comment:
You probably want `(1- n)` rather than `(cl-decf n)`. The latter is side-effecting, and there's no point in that here.

Reply:
Gladly. Used `cl-decf` only because it was there already.